llvm-project/Dockerfile.bootstrap

```diff
@@ -0,0 +1,218 @@
+  -D LLVM_ENABLE_PROJECTS="clang;lld" \
+  -D LLVM_ENABLE_RUNTIMES="compiler-rt;libcxx;libcxxabi;libunwind;" \
+  -D LLVM_DEFAULT_TARGET_TRIPLE=x86_64-alpine-linux-musl \
+  -D CLANG_ENABLE_ARCMT=OFF \
+  -D CLANG_ENABLE_STATIC_ANALYZER=OFF \
+  #-D CLANG_DEFAULT_UNWINDLIB=libunwind \
+  #-D CLANG_DEFAULT_CXX_STDLIB=libc++ \
+  -D CLANG_DEFAULT_LINKER=lld \
+  #-D CLANG_DEFAULT_OBJCOPY=llvm-objcopy \
+  # TODO: this was ignored by runtimes build...\
+  -D CLANG_DEFAULT_RTLIB=compiler-rt \
+  -D CLANG_PLUGIN_SUPPORT=OFF \
+  #-D CLANG_TOOL_AMDGPU_ARCH_BUILD=OFF \
+  #-D CLANG_TOOL_APINOTES_TEST_BUILD=OFF \
+  #-D CLANG_TOOL_ARCMT_TEST_BUILD=OFF \
+  #-D CLANG_TOOL_CLANG_CHECK_BUILD=OFF \
+  #-D CLANG_TOOL_CLANG_DIFF_BUILD=OFF \
+  #-D CLANG_TOOL_CLANG_EXTDEF_MAPPING_BUILD=OFF \
+  #-D CLANG_TOOL_CLANG_FORMAT_BUILD=OFF \
+  #-D CLANG_TOOL_CLANG_FORMAT_VS_BUILD=OFF \
+  #-D CLANG_TOOL_CLANG_FUZZER_BUILD=OFF \
+  #-D CLANG_TOOL_CLANG_IMPORT_TEST_BUILD=OFF \
+  #-D CLANG_TOOL_CLANG_LINKER_WRAPPER_BUILD=OFF \
+  #-D CLANG_TOOL_CLANG_NVLINK_WRAPPER_BUILD=OFF \
+  #-D CLANG_TOOL_CLANG_OFFLOAD_BUNDLER_BUILD=OFF \
+  #-D CLANG_TOOL_CLANG_OFFLOAD_WRAPPER_BUILD=OFF \
+  #-D CLANG_TOOL_CLANG_REFACTOR_BUILD=OFF \
+  #-D CLANG_TOOL_CLANG_RENAME_BUILD=OFF \
+  #-D CLANG_TOOL_CLANG_REPL_BUILD=OFF \
+  #-D CLANG_TOOL_CLANG_SCAN_DEPS_BUILD=OFF \
+  #-D CLANG_TOOL_CLANG_SHLIB_BUILD=OFF \
+  #-D CLANG_TOOL_C_ARCMT_TEST_BUILD=OFF \
+  -D COMPILER_RT_BUILD_MEMPROF=OFF \
+  -D COMPILER_RT_BUILD_BUILTINS=ON \
+  -D COMPILER_RT_BUILD_GWP_ASAN=OFF \
+  -D COMPILER_RT_BUILD_LIBFUZZER=OFF \
+  -D COMPILER_RT_BUILD_ORC=OFF \
+  -D COMPILER_RT_BUILD_PROFILE=OFF \
+  -D COMPILER_RT_BUILD_XRAY=OFF \
+  -D COMPILER_RT_BUILD_SANITIZERS=OFF \
+  -D COMPILER_RT_HAS_GCC_S_LIB=OFF \
+  -D LIBUNWIND_INCLUDE_DOCS=OFF \
+  -D LIBUNWIND_INCLUDE_TESTS=OFF \
+  -D LIBUNWIND_INSTALL_HEADERS=ON \
+  -D LIBUNWIND_USE_COMPILER_RT=ON \
+  -D LIBCXXABI_INCLUDE_TESTS=OFF \
+  -D LIBCXXABI_USE_COMPILER_RT=ON \
+  -D LIBCXXABI_USE_LLVM_UNWINDER=ON \
+  -D LIBCXXABI_ENABLE_STATIC_UNWINDER=ON \
+  -D LIBCXX_HAS_ATOMIC_LIB=OFF \
+  -D LIBCXX_HAS_GCC_LIB=OFF \
+  -D LIBCXX_HAS_GCC_S_LIB=OFF \
+  -D LIBCXX_HAS_MUSL_LIBC=ON \
+  -D LIBCXX_CXX_ABI=libcxxabi \
+  -D LIBCXX_ENABLE_EXPERIMENTAL_LIBRARY=OFF \
+  -D LIBCXX_INCLUDE_BENCHMARKS=OFF \
+  -D LIBCXX_INCLUDE_DOCS=OFF \
+  -D LIBCXX_INCLUDE_TESTS=OFF \
+  -D LIBCXX_USE_COMPILER_RT=ON \
+  -G Ninja \
+  ..
+
+RUN cd llvm-project/llvm/build && \
+  ninja install-builtins
+RUN cd llvm-project/llvm/build && \
+  ninja install-compiler-rt
+RUN cd llvm-project/llvm/build && \
+  ninja install-unwind
+RUN cd llvm-project/llvm/build && \
+  ninja install-cxxabi
+# Necessary for libcxx
+RUN apk add --no-cache linux-headers && \
+  cd llvm-project/llvm/build && \
+  ninja install-cxx
+RUN cd llvm-project/llvm/build && \
+  ninja install-clang install-lld
+RUN cd llvm-project/llvm/build && \
+  ninja install-clang-resource-headers
+
```

Comment:
Can this not be one command?

```
RUN ninja -C llvm-project/llvm/build install-builtins install-compiler-rt ...
```

Reply:
In general, all these can be one big command. That is preferential when distributing said image as it reduces the size (IIUC).

But for development, it's a PITA when one of the sub commands fail, none of the results of a sucessful build are cached. That is a massive PITA when developing docker images, since the dev loop involves multiple painful rebuilds of most of llvm when one part of the chain fails.

Reply:
Ah, good point! Probably worth a comment, since I am sure I am not the only person that will be confused by that.

Reply:
done PTAL